bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/ScenarioHandler.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschshc.internal.devices.bridge;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.Scenario;
+import org.openhab.binding.boschshc.internal.serialization.GsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Handler for executing a scenario.
+ *
+ * @author Patrick Gell - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class ScenarioHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final Map<String, Scenario> availableScenarios;
+
+    protected ScenarioHandler(Map<String, Scenario> availableScenarios) {
+        this.availableScenarios = Objects.requireNonNullElseGet(availableScenarios, HashMap::new);
+    }
+
+    public void executeScenario(final @Nullable BoschHttpClient httpClient, final String scenarioName) {
+        assert httpClient != null;
```

Comment:
Accepting `httpClient` as nullable complicates things, because the resulting object will be in an invalid state and you'll not be able to do anything with it. It would be better to handle this error condition in the handler.

Reply:
Have to make the `httpClient` parameter `@Nullable` because in the `BridgeHandler` class it is also defined as `@Nullable`. I could make a null check instead of the assert and than simply return from this function.   
Or is there another way how this is handled in OH?

Reply:
You already have a null check in `BridgeHandler` before calling the method. So you can remove `@Nullable` from the parameter and all null checks in this method and the private method called.

Reply:
But if I remove the `@Nullable` I always get a compile error even though there is the null check in the calling method.

```java
Null type mismatch (type annotations): required 'org.openhab.binding.boschshc.internal.devices.bridge.@NonNull BoschHttpClient' but this expression has type 'org.openhab.binding.boschshc.internal.devices.bridge.@Nullable BoschHttpClient'
```

Reply:
In which class do you get this error? It seems to be a null annotation issue in `BridgeHandler` - I'll comment directly there.

Reply:
`@Nullable`was removed